File: src/main/java/de/churl/cookbook/domain/model/IngredientUsageKey.java
package de.churl.cookbook.domain.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import java.io.Serializable;
import java.util.Objects;
import java.util.UUID;

@Embeddable
@Getter
@Setter
@NoArgsConstructor
public class IngredientUsageKey implements Serializable {

    @Column(name = "recipe_id")
    private UUID recipeID;

    @Column(name = "ingr_id")
    private UUID ingrID;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || this.getClass() != o.getClass()) {
            return false;
        }
        final IngredientUsageKey that = (IngredientUsageKey) o;
        return this.recipeID.equals(that.recipeID) && this.ingrID.equals(that.ingrID);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.recipeID, this.ingrID);
    }
}

File: src/main/kotlin/de/churl/cookbook/model/IngredientUsageKey.kt
package de.churl.cookbook.model

import java.io.Serializable
import java.util.*
import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
data class IngredientUsageKey (
        @field: Column(name = "recipe_id") var recipeID : UUID,
        @field: Column(name = "ingr_id") var ingrID : UUID
        ) : Serializable
